### PR TITLE
Fixes insensitive for DESC table for encrypt.

### LINKED ISSUE #4770: Case insensitive for DESC table for encrypt
DESC table is case sensitive now, We should change it to case insensitive.

### PR DESCRIPTION
**Pull request by dongzl**
Fixes #4770 .

Changes proposed in this pull request:
- Get column encrypt field from SchemaMetaData instead of EncryptRule.

### CODE REVIEW

File: encrypt-core/encrypt-core-merge/src/main/java/org/apache/shardingsphere/encrypt/merge/dal/impl/EncryptColumnsMergedResult.java [hunk 1]
```diff
@@ -66,19 +70,50 @@ public final boolean next() throws SQLException {
...
+    }
+    
+    private Collection<String> getPlainColumns() {
+        return getTableEncryptColumnMetaDatas().stream().map(EncryptColumnMetaData::getPlainColumnName)
+                .collect(Collectors.toList());
+    }
+    
+    private Collection<EncryptColumnMetaData> getTableEncryptColumnMetaDatas() {
+        Collection<EncryptColumnMetaData> result = new LinkedList<>();
+        for (Map.Entry<String, ColumnMetaData> entry : schemaMetaData.get(tableName).getColumns().entrySet()) {
```
Comment (terrymanu) [inline encrypt-core/encrypt-core-merge/src/main/java/org/apache/shardingsphere/encrypt/merge/dal/impl/EncryptColumnsMergedResult.java]: Please do not use `Map.Entry`, just use `Entry`

File: encrypt-core/encrypt-core-merge/src/main/java/org/apache/shardingsphere/encrypt/merge/dal/impl/EncryptColumnsMergedResult.java [hunk 2]
```diff
@@ -66,19 +70,50 @@ public final boolean next() throws SQLException {
...
+            if (!(entry.getValue() instanceof EncryptColumnMetaData)) {
+                continue;
+            }
+            result.add((EncryptColumnMetaData) entry.getValue());
```
Comment (terrymanu) [inline encrypt-core/encrypt-core-merge/src/main/java/org/apache/shardingsphere/encrypt/merge/dal/impl/EncryptColumnsMergedResult.java]: It is better to change to: 

```java
if (entry.getValue() instanceof EncryptColumnMetaData) {
    result.add((EncryptColumnMetaData) entry.getValue());
}
```

File: encrypt-core/encrypt-core-merge/src/main/java/org/apache/shardingsphere/encrypt/merge/dal/impl/EncryptColumnsMergedResult.java [hunk 3]
```diff
@@ -66,19 +70,50 @@ public final boolean next() throws SQLException {
...
-            }
-            return columnName;
+            Optional<String> logicColumn = getLogicColumnOfCipher(columnName);
+            return logicColumn.isPresent() ? logicColumn.get() : columnName;
         }
         return getOriginalValue(columnIndex, type);
     }
     
+    private Optional<String> getLogicColumnOfCipher(final String cipherColumn) {
+        for (Map.Entry<String, ColumnMetaData> entry : schemaMetaData.get(tableName).getColumns().entrySet()) {
```
Comment (terrymanu) [inline encrypt-core/encrypt-core-merge/src/main/java/org/apache/shardingsphere/encrypt/merge/dal/impl/EncryptColumnsMergedResult.java]: Please do not use `Map.Entry`, just use `Entry`

File: encrypt-core/encrypt-core-merge/src/main/java/org/apache/shardingsphere/encrypt/merge/dal/impl/EncryptColumnsMergedResult.java (hunk 1 above)
Comment (dongzl) [reply encrypt-core/encrypt-core-merge/src/main/java/org/apache/shardingsphere/encrypt/merge/dal/impl/EncryptColumnsMergedResult.java]: OK, I fix it. Thanks.

File: encrypt-core/encrypt-core-merge/src/main/java/org/apache/shardingsphere/encrypt/merge/dal/impl/EncryptColumnsMergedResult.java (hunk 2 above)
Comment (dongzl) [reply encrypt-core/encrypt-core-merge/src/main/java/org/apache/shardingsphere/encrypt/merge/dal/impl/EncryptColumnsMergedResult.java]: OK, I fix it. Thanks.

**Pull request merged**